pkg/agent/datamodel/types.go

```diff
@@ -1286,3 +1293,395 @@ func FormatProdFQDNByLocation(fqdnPrefix string, location string, cloudSpecConfi
 	FQDNFormat := cloudSpecConfig.EndpointConfig.ResourceManagerVMDNSSuffix
 	return fmt.Sprintf("%s.%s."+FQDNFormat, fqdnPrefix, location)
 }
+
+type K8sComponents struct {
+	// Full path to the "pause" image. Used for --pod-infra-container-image
+	// For example: "mcr.microsoft.com/oss/kubernetes/pause:1.3.1"
+	PodInfraContainerImageURL string
+
+	// Full path to the hyperkube image.
+	// For example: "mcr.microsoft.com/hyperkube-amd64:v1.16.13"
+	HyperkubeImageURL string
+
+	// Full path to the Windows package (windowszip) to use.
+	// For example: https://acs-mirror.azureedge.net/kubernetes/v1.17.8/windowszip/v1.17.8-1int.zip
+	WindowsPackageURL string
+}
+
+// NodeBootstrappingConfiguration represents configurations for node bootstrapping
+type NodeBootstrappingConfiguration struct {
+	ContainerService              *ContainerService
+	CloudSpecConfig               *AzureEnvironmentSpecConfig
+	K8sComponents                 *K8sComponents
+	AgentPoolProfile              *AgentPoolProfile
+	TenantID                      string
+	SubscriptionID                string
+	ResourceGroupName             string
+	UserAssignedIdentityClientID  string
+	ConfigGPUDriverIfNeeded       bool
+	EnableGPUDevicePluginIfNeeded bool
+	EnableDynamicKubelet          bool
```

Comment:
@bowang-666  FYI potential conflict with your change that renames this property.

Reply:
Just merged this https://github.com/Azure/AgentBaker/pull/411
Could you help rename property name here?

Reply:
I have resolved merge conflict with #411. Please take another look.